dnf-plugins/py3query.py

```diff
@@ -174,16 +174,31 @@ def set_status(result, pkgs, python_versions):
         else:
             result['status'] = 'idle'
 
-    set_py2status(result, pkg_by_version)
+    check_naming_policy(result, pkg_by_version, name_by_version)
 
 
-def set_py2status(result, pkg_by_version):
+def check_naming_policy(result, pkg_by_version, name_by_version):
     """Check if Python 2 subpackages are correctly named."""
-    result['py2status'] = 'released'
     for pkg in pkg_by_version[2]:
-        if pkg.name.startswith('python-') or pkg.name.endswith('-python'):
-            result['py2status'] = 'mispackaged'
-            break
+        # Unversioned python is used in package name (e.g. python-foo).
+        unversioned_prefix = (
+            pkg.name.startswith('python-') or
+            '-python-' in pkg.name or
+            pkg.name.endswith('-python'))
+
+        # Missing python2- prefix (e.g. foo and python3-foo).
+        missing_prefix = (
+            'python' not in pkg.name and
+            # There is a python3-foo package.
+            ('python3-{}'.format(pkg.name) in name_by_version[3] or
+             '{}-python3'.format(pkg.name) in name_by_version[3]) and
+            # There is no python2-foo package.
+            ('python2-{}'.format(pkg.name) not in name_by_version[2] and
+             '{}-python2'.format(pkg.name) not in name_by_version[2]))
```

Comment:
Yes, this should work and will look better. Thanks for a suggestion, will change it.